test/framework/management_cluster.go

```diff
@@ -158,3 +166,87 @@ func WaitForDeploymentsAvailable(ctx context.Context, input WaitForDeploymentsAv
 
 	}, intervals...).Should(BeTrue(), "Deployment %s/%s failed to get status.Available = True condition", input.Deployment.GetNamespace(), input.Deployment.GetName())
 }
+
+// CreateNamespaceInput is the input type for CreateNamespace.
+type CreateNamespaceInput struct {
+	Creator Creator
+	Name    string
```

Comment:
Any particular reason to require a `Name` on input for CreateNamespace?

Could you make `Name` optional  and default to `"test-" + util.RandomString(6)` if not provided? Otherwise downstream providers consuming this would still have to implement their own randomization.

Reply:
informatively, the k/k framework:
(in the default Before function)
- takes the test name as input
- adds a UUID after the test name
- loops to make sure a namespace name+uuid does not exist
- creates the namespace
(in the default After function)
- deletes the namespace

Reply:
@detiber Done!
@neolit123 yes, we are following a similar path